WebDriverAgentLib/Categories/XCUIElement+FBFind.m

```diff
@@ -69,51 +80,87 @@ - (void)descendantsWithProperty:(NSString *)property value:(NSString *)value par
 
 #pragma mark - Search by Predicate String
 
-- (NSArray<XCUIElement *> *)fb_descendantsMatchingPredicate:(NSPredicate *)predicate {
+- (NSArray<XCUIElement *> *)fb_descendantsMatchingPredicate:(NSPredicate *)predicate shouldReturnAfterFirstMatch:(BOOL)shouldReturnAfterFirstMatch
+{
   XCUIElementQuery *query = [[self descendantsMatchingType:XCUIElementTypeAny] matchingPredicate:predicate];
-  NSArray *childElements = [query allElementsBoundByIndex];
-  return childElements;
+  NSMutableArray *result = [NSMutableArray array];
```

Comment:
I really don't like repetition. You should extract this code to helper method in that class.